File: Easygrants_Web_LSC/Easygrants_Web/EG_Core_2/Easygrants/EasyGrants/Web/User/EGHomeUser.cs
using System;
using System.Diagnostics;
using System.Collections;
using System.Xml;
using System.Data;
using Core.DataAccess;
using Core.DataAccess.DBAccess;
using Core.DataAccess.XMLAccess;
using Core.Web.User;
using EasyGrants.Workflow;

namespace EasyGrants.Web.User {
/*====================================================*/

///<summary>
///user in home module
///</summary>
public class cEGHomeUser : cEGUser 
{
/*====================================================*/

	public cEGHomeUser(cEGAppl aAppl, cEGSession aSession) :
					base(aAppl, aSession) {
	}
	
/*-----------------------------------------------------*/	

	public cEGHomeUser(cEGUser aUser) :
					base(aUser) {

	}

/*=====================================================*/

	public override void Notify_DataModified(cDataObjectFactory aDataObjectFactory, Hashtable aDataObjHash) {
		cXMLDoc				aXmlDoc;
		XmlNode				aDataObjectNode;
		XmlNode				aFiltersNode;
		XmlNode				aArgNode;
		cDataObjectList		aWfTaskAssignments;
		
		aXmlDoc = aDataObjectFactory.CreateDataObjectNode("WfTaskAssignment", out aDataObjectNode, out aFiltersNode);	
		aArgNode = aFiltersNode.AppendChild(aDataObjectFactory.CreateArgumentNode(aXmlDoc, "Data",
											"PersonID", this.PersonID.ToString()));
		
		aWfTaskAssignments = aDataObjectFactory.GetDataObjectList(aDataObjectNode);
		foreach(cWfTaskAssignment aWfTaskAssignment in aWfTaskAssignments) {
			Debug.WriteLine("ID, Description: " + aWfTaskAssignment.GetPropertyString("WfTaskAssignmentID") + ", " +
							aWfTaskAssignment.GetRelatedPropertyString("WfTask.WfTask") + " (as " +
							aWfTaskAssignment.GetRelatedPropertyString("WfTaskRole.WfTaskRole") + ")");
			
			aWfTaskAssignment.Notify_DataModified(aDataObjectFactory, aDataObjHash);
		}
	}

/*=====================================================*/
} // class cEGHomeUser


/*====================================================*/
}  //namespace EasyGrants.Web.User

File: Easygrants_Web_LSC/Easygrants_Web_LSC/Easygrants/User/EGHomeUser.vb
Imports Core.DataAccess
Imports Core.Web.User
Imports EasyGrants.Web
Imports EasyGrants.Web.User

Namespace EasyGrants_Web.User

Public Class cEGHomeUser
	Inherits EasyGrants.Web.User.cEGHomeUser

	Public Sub New(ByVal aAppl As cEGAppl, ByVal aSession As cEGSession)
		MyBase.New(aAppl, aSession)
	End Sub


	Public Sub New(ByVal aUser As cEGUser)
		MyBase.New(aUser)
	End Sub

	Public Overrides Sub Notify_DataModified(ByVal aDataObjectFactory As cDataObjectFactory, ByVal aDataObjHash As Hashtable)
		'Do nothing on save from the home module. 
		'This overrides the baseline functionality which automatically wipes out all PDFs corresponding to a user when the user saves any data in the Home module.
	End Sub

End Class

End Namespace
